modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+    if (hasMeshProperty("node_id_background_meta", name() + "_circle"))
+      declareMeshProperty(
+          "node_id_background_meta",
+          getMeshProperty<unsigned int>("node_id_background_meta", name() + "_circle"));
+    if (hasMeshProperty("pattern_pitch_meta", name() + "_circle"))
+      declareMeshProperty("pattern_pitch_meta",
+                          getMeshProperty<Real>("pattern_pitch_meta", name() + "_circle"));
+    if (hasMeshProperty("azimuthal_angle_meta", name() + "_circle"))
+      declareMeshProperty(
+          "azimuthal_angle_meta",
+          getMeshProperty<std::vector<Real>>("azimuthal_angle_meta", name() + "_circle"));
+    if (hasMeshProperty("num_sectors_per_side_meta", name() + "_circle"))
+      declareMeshProperty("num_sectors_per_side_meta",
+                          getMeshProperty<std::vector<unsigned int>>("num_sectors_per_side_meta",
+                                                                     name() + "_circle"));
+    if (hasMeshProperty("max_radius_meta", name() + "_circle"))
+      declareMeshProperty("max_radius_meta",
+                          getMeshProperty<Real>("max_radius_meta", name() + "_circle"));
+
+    // Define boundary IDs so that there are unified boundaries for use in patterned MeshGenerator
+    // if this pin is to be used in an AssemblyMeshGenerator
+    {
+      auto params = _app.getFactory().getValidParams("SideSetsFromNormalsGenerator");
+
+      params.set<MeshGeneratorName>("input") = name() + "trans";
+      params.set<std::vector<Point>>("normals") = {
+          {1, 0, 0},
+          {-1, 0, 0},
+          {0, 1, 0},
+          {0, -1, 0}}; // normal directions over which to define boundaries
+      params.set<bool>("fixed_normal") = true;
+      params.set<bool>("replace") = false;
+      params.set<std::vector<BoundaryName>>("new_boundary") = {"10001", "10002", "10003", "10004"};
+
+      _build_mesh = &addMeshSubgenerator("SideSetsFromNormalsGenerator", name() + "_2D", params);
+    }
+  }
+  else
+  {
+    // Hex geometry
+    {
+      auto params = _app.getFactory().getValidParams("PolygonConcentricCircleMeshGenerator");
+      params.set<unsigned int>("num_sides") = 6; // Cartesian geometry so pin are given a square box
+      params.set<std::vector<unsigned int>>("num_sectors_per_side") =
+          std::vector<unsigned int>(6, _num_sectors);
+      params.set<bool>("preserve_volumes") = true;
+      params.set<bool>("quad_center_elements") = _quad_center;
+      params.set<boundary_id_type>("external_boundary_id") =
+          20000 +
+          _pin_type; // The boundary id for pins as set by the Reactor Geometry Meshing convention
+      params.set<std::string>("external_boundary_name") = "outer_pin_" + std::to_string(_pin_type);
+
+      // Break the mesh intervals and region ids up by construct for
+      // PolygonConcentricCircleMeshGenerator
+      std::vector<unsigned int> ring_intervals;
+      std::vector<subdomain_id_type> ring_ids;
```

Comment:
rename to ring_blk_ids since the future mesh attribute added to rings will be `ring_id` and it could get confusiing